### PR TITLE
changes in ui to get the step error visible in ui when dry run runs the exectional script

### PR DESCRIPTION
**Pull request by NJ-jain**
reviewed myself

### CODE REVIEW

**Comment by NJ-jain** [comment]
pleasee check this pr on your  system and check the ui for this also !!
and for runnig this branch need to have backend branch named "change-steporder-to-json" and add env -
WRITE_INDEX= local_json_alias_1
READ_INDEX = local_json_alias

File: src/components/functiondashboard/workFlow/workFlowComponent/apiFunctionComponent/apiFunctionComponent.js [hunk 1]
```diff
@@ -13,9 +13,11 @@ import { handleDeleteStepsThunk } from '../../../../../store/flowJson/flowJsonTh
 import { updateAppInfoState } from '../../../../../store/appInfo/appInfoSlice.ts'
 import { Tabnames } from '../../../../../enums'
 import UsedVariable from '../usedVariableComponent/usedVariables.tsx'
+import { useCustomSelector } from '../../../../../utils/deepCheckSelector'
 
 export default function ApiFunctionComponent(props) {
   const id = props?.id
+  const blockError = useCustomSelector((state) => state.invocationV2.invocationData.stepOrder[id]?.success) ?? true
```
Comment (SiDDhartharG) [inline src/components/functiondashboard/workFlow/workFlowComponent/apiFunctionComponent/apiFunctionComponent.js:20]: const blockError = useCustomSelector((state) => state.invocationV2.invocationData.stepOrder?[id]?.success) ?? true

File: src/components/functiondashboard/workFlow/workFlowComponent/ifComponent/ifComponent.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ import { evalVariableAndCodeFromContext } from '../../../../../utils/codeUtility
 function IfComponent({ stepId, closedIfBlocks }) {
   const dispatch = useDispatch()
   const stepBlocks = useCustomSelector((state) => state.flowJsonV2?.flowJson?.blocks[stepId])
+  const blockError = useCustomSelector((state) => state.invocationV2?.invocationData?.stepOrder[stepId]?.success) ?? true
```
Comment (SiDDhartharG) [inline src/components/functiondashboard/workFlow/workFlowComponent/ifComponent/ifComponent.js:21]: const blockError = useCustomSelector((state) => state.invocationV2.invocationData.stepOrder?[id]?.success) ?? true

File: src/components/functiondashboard/workFlow/workFlowComponent/pluginComponent/pluginComponent.js [hunk 3]
```diff
@@ -13,9 +13,11 @@ import { BlockTypes, Tabnames } from '../../../../../enums'
 import UsedVariable from '../usedVariableComponent/usedVariables.tsx'
 
 import { resetCurrentSelectedPlugin } from '../../../../../store/plugin/currentSelectedPlugin/currentSelectedPlugin.ts'
+import { useCustomSelector } from '../../../../../utils/deepCheckSelector'
 
 export default function PluginComponent({ id, ifId, duplicateFunctionApiComponent }) {
   const stepInfo = useSelector((state) => state.flowJsonV2.flowJson.blocks[id])
+  const blockError = useCustomSelector((state) => state.invocationV2.invocationData.stepOrder[id]?.success) ?? true
```
Comment (SiDDhartharG) [inline src/components/functiondashboard/workFlow/workFlowComponent/pluginComponent/pluginComponent.js:20]: const blockError = useCustomSelector((state) => state.invocationV2.invocationData.stepOrder?[id]?.success) ?? true

**Review by SiDDhartharG — CHANGES_REQUESTED** [review]

**Review by SiDDhartharG — APPROVED** [review]

**Pull request closed without merging**